imitation/airl.py

```diff
@@ -31,75 +37,92 @@ def __init__(self, env, gen_policy, discrim, expert_policies, *,
     self.expert_policies = expert_policies
-    self.init_tensorboard = init_tensorboard
+    self._n_disc_samples_per_buffer = n_disc_samples_per_buffer
 
     self._global_step = tf.train.create_global_step()
 
     with tf.variable_scope("AIRLTrainer"):
       with tf.variable_scope("discriminator"):
         self.discrim = discrim
-
         self._build_disc_train()
       self._build_policy_train_reward()
       self._build_test_reward()
-
-    if self.init_tensorboard:
+    self._init_tensorboard = init_tensorboard
+    if init_tensorboard:
       with tf.name_scope("summaries"):
         self._build_summarize()
 
     self._sess.run(tf.global_variables_initializer())
 
     self.env = self.wrap_env_train_reward(self.env)
+    self.gen_policy.set_env(self.env)
+
+    gen_replay_buffer_capacity = 20 * self._n_disc_samples_per_buffer
+    self._gen_replay_buffer = ReplayBuffer(gen_replay_buffer_capacity, self.env)
+    self._exp_replay_buffer = ReplayBuffer.fromData(
+      *util.rollout.generate_multiple(
+        self.expert_policies, self.env, n_expert_samples)[:3])
+    self._populate_gen_replay_buffer()
 
   def train_disc(self, *, n_steps=10, **kwargs):
     """Trains the discriminator to minimize classification cross-entropy.
 
-    The generator rollout parameters of the form "gen_*" are optional,
-    but if one is given, then all such parameters must be filled (otherwise
-    this method errors). If none of the generator rollout parameters are
-    given, then a rollout with the same length as the expert rollout
-    is generated on the fly.
-
     Args:
         n_steps (int): The number of training steps taken.
-        gen_old_obs (array): A numpy array with shape
-            `[n_timesteps] + env.observation_space.shape`. The ith observation
-            in this array should be the observation seen when the generator
-            chooses action `gen_act[i]`.
-        gen_act (array): A numpy array with shape
-            `[n_timesteps] + env.action_space.shape`.
-        gen_new_obs (array): A numpy array with shape
-            `[n_timesteps] + env.observation_space.shape`. The ith observation
-            in this array is from the transition state after the generator
-            chooses action `gen_act[i]`.
     """
-    fd = self._build_disc_feed_dict(**kwargs)
     for _ in range(n_steps):
+      fd = self._build_disc_feed_dict(**kwargs)
       step, _ = self._sess.run([self._global_step, self._disc_train_op],
                                feed_dict=fd)
-      if self.init_tensorboard and step % 20 == 0:
+      if self._init_tensorboard and step % 20 == 0:
         self._summarize(fd, step)
 
   def train_gen(self, n_steps=10000):
     self.gen_policy.set_env(self.env)
     self.gen_policy.learn(n_steps)
+    self._populate_gen_replay_buffer()
+
+  def _populate_gen_replay_buffer(self) -> None:
+    """Generate and store `self._n_disc_sample_per_buffer` new generator
```

Comment:
I'm not sure how this is different from Google format. Is it because it's lacking a multi-line summary?

(Love google docstring format, just ocassionally forget when I'm thinking hard about docstring wording)

Reply:
Gotcha, saw your other comment.

Reply:
"A docstring should be organized as a summary line (one physical line) terminated by a period, question mark, or exclamation point, followed by a blank line, followed by the rest of the docstring starting at the same cursor position as the first quote of the first line." - http://google.github.io/styleguide/pyguide.html#381-docstrings

So the summary needs to just be one line, this spans over two lines.